frontend/app/src/shared/components/form/utils/form-context.tsx

```diff
@@ -0,0 +1,26 @@
+import { NodeObject } from "@/entities/nodes/types";
+import { NodeSchema } from "@/entities/schema/types";
+import { ReactNode, createContext } from "react";
+
+export interface FormContextType {
+  parentSchema: NodeSchema | null;
+  parentData: NodeObject | null;
+}
+
+export interface FormContextProps extends FormContextType {
+  children: ReactNode;
+}
+
+export const FormContext = createContext<FormContextType>({
```

Comment:
I'm wondering if ParentContext is a more explicit alternative? WDYT.
(btw we also have FormFieldContext to handle each form field state)